### PR TITLE
fix: react hooks observers

### PR DESCRIPTION
**Pull request by helciofranco**
This PR fixes and reduces some re-renders across our query observers.

This issue was caused due to the way which `@tanstack/react-query` handles the tracked properties to re-renders.
Internally, they implemented a `Proxy` to intercept and track which properties a component is listening from a `useQuery` hook.

I'm importing our `useNamedQuery` to keep the API backwards compatible.

### CODE REVIEW

File: packages/app-portal/package.json [hunk 1]
```diff
@@ -16,14 +16,13 @@
     "@fuel-bridge/fungible-token": "next",
     "@fuel-bridge/message-predicates": "next",
     "@fuel-bridge/solidity-contracts": "next",
-    "@fuel-wallet/react": "0.15.2",
+    "@fuel-wallet/react": "file:/Users/helciofranco/Documents/wallet/packages/react",
```
Comment (helciofranco) [inline packages/app-portal/package.json:19]: Waiting on https://github.com/FuelLabs/fuels-wallet/pull/1127

File: packages/app-portal/src/providers.tsx [hunk 2]
```diff
@@ -12,13 +10,9 @@ type ProvidersProps = {
 export function Providers({ children }: ProvidersProps) {
   return (
     <StoreProvider>
-      <FuelProvider>
```
Comment (helciofranco) [inline packages/app-portal/src/providers.tsx:15]: `FuelProvider` was duplicated here. 
`FuelConnectProvider` already have that.

File: packages/app-portal/src/providers.tsx [hunk 3]
```diff
@@ -12,13 +10,9 @@ type ProvidersProps = {
...
-        <FuelConnectProvider>
-          <ConnectProvider>
-            <FuelNetworkProvider>{children}</FuelNetworkProvider>
```
Comment (helciofranco) [inline packages/app-portal/src/providers.tsx:18]: `FuelNetworkProvider` was transferring react-query's data to context.
I just removed that context in order to keep server-state data focused on react-query.

File: packages/app-portal/package.json [hunk 4]
```diff
@@ -41,9 +40,7 @@
   },
   "devDependencies": {
     "@storybook/react": "^7.6.10",
-    "@tanstack/react-query": "4.36.1",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "@tanstack/react-query-devtools": "4.36.1",
```
Comment (helciofranco) [inline packages/app-portal/package.json:43]: this is helpful to debug what our `react-query` has in cache
(no worry with the bundle size, it's always removed from production bundles automatically)

**Review by luizstacio — APPROVED** [review]

File: packages/app-portal/package.json [hunk 5]
```diff
@@ -16,14 +16,13 @@
     "@fuel-bridge/fungible-token": "next",
     "@fuel-bridge/message-predicates": "next",
     "@fuel-bridge/solidity-contracts": "next",
-    "@fuel-wallet/react": "0.15.2",
+    "@fuel-wallet/react": "0.16.0",
```
Comment (helciofranco) [inline packages/app-portal/package.json:19]: this version has the changes to re-render only tracked properties

**Pull request merged**